source/groupBy.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+/**
+ * Функция, группирующая объекты из массива по типу
+ * @param {Array<Objects>} input - массив объектов
+ * @param key - ключ, по которому группируются элементы массива 
+ * 
+ * @returns {type: {Objects}}
+ */
+const groupBy = (input, key) => {
+    if (!Array.isArray(input)) () => { throw new TypeError('Первый аргумент должен быть массивом объектов'); }
```

Comment:
Странная строчка, может так должно быть?
```suggestion
    if (!Array.isArray(input)) {
        throw new TypeError('Первый аргумент должен быть массивом объектов');
    }
```